app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.java

```diff
@@ -159,220 +193,173 @@ public void readFrom(@NonNull Queue<Object> savedObjects) throws Exception {
-     * <p>
-     * On initialization, it automatically requests the amount of feed needed to display
-     * a minimum amount required (FEED_LOAD_SIZE).
-     * <p>
-     * Upon receiving a user pull, it creates a Single Observer to fetch the ChannelInfo
-     * containing the feed streams.
-     **/
-    private Subscriber<SubscriptionEntity> getSubscriptionObserver() {
-        return new Subscriber<SubscriptionEntity>() {
-            @Override
-            public void onSubscribe(Subscription s) {
-                if (feedSubscriber != null) feedSubscriber.cancel();
-                feedSubscriber = s;
+        for (SubscriptionEntity subscriptionEntity: result) {
+            if (loadedSubscriptionEntities.contains(subscriptionEntity.getUrl())) continue;
+            filteredResult.add(subscriptionEntity);
+        }
 
-                int requestSize = FEED_LOAD_COUNT - infoListAdapter.getItemsList().size();
-                if (wasLoading.getAndSet(false)) requestSize = FEED_LOAD_COUNT;
+        numChannels.set(filteredResult.size());
+        compositeDisposable.add(
+            Flowable.fromIterable(filteredResult)
+                .observeOn(Schedulers.io())
+                .subscribe(this::handleReceiveSubscriptionEntity, this::handleError));
 
-                boolean hasToLoad = requestSize > 0;
-                if (hasToLoad) {
-                    requestLoadedAtomic.set(infoListAdapter.getItemsList().size());
-                    requestFeed(requestSize);
-                }
-                isLoading.set(hasToLoad);
-            }
+        // Start item list UI update scheduler
+        delayHandler.postDelayed(this::updateItemsList, 3200);
+    }
 
-            @Override
-            public void onNext(SubscriptionEntity subscriptionEntity) {
-                if (!itemsLoaded.contains(subscriptionEntity.getServiceId() + subscriptionEntity.getUrl())) {
-                    subscriptionService.getChannelInfo(subscriptionEntity)
-                            .observeOn(AndroidSchedulers.mainThread())
-                            .onErrorComplete(
-                                    (@io.reactivex.annotations.NonNull Throwable throwable) ->
-                                            FeedFragment.super.onError(throwable))
-                            .subscribe(
-                                    getChannelInfoObserver(subscriptionEntity.getServiceId(),
-                                            subscriptionEntity.getUrl()));
-                } else {
-                    requestFeed(1);
-                }
-            }
+    private void handleReceiveSubscriptionEntity(SubscriptionEntity subscriptionEntity) {
+        compositeDisposable.add(
+            subscriptionService.getChannelInfo(subscriptionEntity)
+                .observeOn(Schedulers.io())
+                .onErrorComplete((@io.reactivex.annotations.NonNull Throwable throwable) -> FeedFragment.super.onError(
```

Comment:
Can't `@io.reactivex.annotations.NonNull` be imported and referenced just as `@NonNull`?